util/threadpool.cc

```diff
@@ -23,29 +28,128 @@ void ThreadPool::PthreadCall(const char* label, int result) {
   }
 }
 
+namespace {
+#ifdef ROCKSDB_STD_THREADPOOL
+
+struct Lock {
```

Comment:
@ajkr Yes, we follow the same paths as pthread locks, except the exception path which std::condition_variable claims to throw if the mutex is not owned.
On a slightly more practical note, std::condition_variable requires unique_lock for waiting on the lock. This is the way it is written. I chose to create one set of wrappers for all of the cases for simplicity even when waiting is not required.